### ISSUE TITLE
Unable to manage tenants through Management Console (BC17.0)

### ISSUE THREAD

**Issue by krregg** [MicrosoftDocs/dynamics365smb-devitpro-pb#1523]
I am working with the Business Central version 17.0.

I just performed a migration to multitenant. I used Powershell commands to separate application and mount tenant. All works fine no error, service is accessible. 

Now I tried to manage tenants through Management Console, where I choose NST instance on the left first, and then when I try to choose 'Tenants' I get an error.:

```
FX:{ba484c42-ed9a-0170-925f-23e64e686fd0}

Value cannot be null.
Parameter name: objectToValidate

at Microsoft.ManagementConsole.BaseCollection.OnValidate(Object objectToValidate, Boolean testForDuplicate)
   at System.Collections.CollectionBase.System.Collections.IList.Add(Object value)
   at CallSite.Target(Closure , CallSite , NodeSubItemDisplayNameCollection , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
   at Microsoft.Dynamics.Nav.ManagementUI.NavTenantsListView.RefreshWithNodeStateChangedCheck(IList`1 tenants, ResultNodeCollection currentResultNodes)
   at Microsoft.Dynamics.Nav.ManagementUI.NavTenantsListView.RefreshCore()
   at Microsoft.Dynamics.Nav.ManagementUI.NavTenantsListView.OnInitialize(AsyncStatus status)
   at Microsoft.ManagementConsole.View.ProcessRequest(Request request)
   at Microsoft.ManagementConsole.ViewMessageClient.ProcessRequest(Request request)
   at Microsoft.ManagementConsole.Internal.IMessageClient.ProcessRequest(Request request)
   at Microsoft.ManagementConsole.Executive.RequestStatus.BeginRequest(IMessageClient messageClient, RequestInfo requestInfo)
   at Microsoft.ManagementConsole.Executive.SnapInRequestOperation.ProcessRequest()
   at Microsoft.ManagementConsole.Executive.Operation.OnThreadTransfer(SimpleOperationCallback callback)
```

I can still manage tenants with Powershell but it seems odd that Management Console dies like that. I was looking at Event Viewer and nothing is there that would be helpful. I can't even find a record of that error.

I am attaching a print screen regarding this error:

[image_1](https://ibb.co/HGcy6Qk)

What should be done to get the tenant visible in Management Console with no error?

Below are the steps I used to convert DB to multitenant:

```
Stop-NAVServerInstance –ServerInstance $serverInstance

Export-NAVApplication –DatabaseServer $db_server `
                      –DatabaseInstance $db_instance `
                      –DatabaseName $db_origin `
                      –DestinationDatabaseName $db_dest_central
                      
Remove-NAVApplication –DatabaseServer $db_server `
                      -DatabaseInstance $db_instance `
                      –DatabaseName $db_origin `
                      -Force

Set-NAVServerConfiguration –ServerInstance $serverInstance `
                           –element appSettings `
                           –KeyName 'DatabaseName' `
                           –KeyValue '' `
                           -Force

Restart-NAVServerInstance –ServerInstance $serverInstance

Mount-NAVApplication –ServerInstance $serverInstance `
                     –DatabaseServer '$db_server\$db_instance' `
                     –DatabaseName $db_dest_central `
                     -Force

Mount-NAVTenant –ServerInstance $serverInstance `
                -Id 'development' `
                –DatabaseServer '$db_server\$db_instance' `
                -DatabaseName $db_origin `
                -OverwriteTenantIdInDatabase

Restart-NAVServerInstance –ServerInstance $serverInstance
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 05d21543-2d25-b705-e3c6-4e2df0054b64
* Version Independent ID: da02181d-08c6-a4e6-9d5f-f185b16f9896
* Content: [Migrating to Multitenancy - Business Central](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/deployment/migrating-to-multitenancy)
* Content Source: [dev-itpro/deployment/Migrating-to-Multitenancy.md](https://github.com/MicrosoftDocs/dynamics365smb-devitpro-pb/blob/live/dev-itpro/deployment/Migrating-to-Multitenancy.md)
* Service: **dynamics365-business-central**
* GitHub Login: @jswymer
* Microsoft Alias: **solsen**

**Comment by jswymer** [comment]
Hello, thanks for the feedback. This seems to be an error in version 17. A defect has been registered and we will investigate.

**Comment by krregg** [comment]
@jswymer, thank you for your reply. I forgot to mention, I tested this case also on the default Cronus database and the result was the same. Everything works fine, only the error in the console still persist.

**Comment by jswymer** [comment]
Hello,

This defect should be fixed in the next update (target 17.2), so I will close this issue. Thanks for the feedback!
